Xamarin.Forms.Maps.Android/MapRenderer.cs

```diff
@@ -291,30 +310,40 @@ void MapOnMarkerClick (object sender, GoogleMap.InfoWindowClickEventArgs eventAr
 
 			// only consider event handled if a handler is present. 
 			// Else allow default behavior of displaying an info window.
-			targetPin?.SendTap ();
+			targetPin?.SendTap();
 		}
 
-		bool _disposed;
-		protected override void Dispose (bool disposing)
+		protected override void Dispose(bool disposing)
 		{
-			if (disposing && !_disposed) {
+			if (_disposed)
+			{
+				return;
+			}
+
+			if (disposing)
+			{
 				_disposed = true;
-			
-				var mapModel = Element as Map;
-				if (mapModel != null) {
-					MessagingCenter.Unsubscribe<Map, MapSpan> (this, MoveMessageName);
-					((ObservableCollection<Pin>)mapModel.Pins).CollectionChanged -= OnCollectionChanged;
+
+				if (Element != null)
+				{
+					MessagingCenter.Unsubscribe<Map, MapSpan>(this, MoveMessageName);
+					((ObservableCollection<Pin>)Element.Pins).CollectionChanged -= OnCollectionChanged;
 				}
 
-				var gmap = NativeMap;
-				if (gmap == null)
-					return;
-				gmap.MyLocationEnabled = false;
-				gmap.InfoWindowClick -= MapOnMarkerClick;
-				gmap.Dispose ();
+				if (NativeMap != null)
+				{
+					NativeMap.MyLocationEnabled = false;
+					NativeMap.SetOnCameraChangeListener(null);
+					NativeMap.InfoWindowClick -= MapOnMarkerClick;
+					NativeMap.Dispose();
+				}
+
+				Control?.OnDestroy();
 			}
 
-			base.Dispose (disposing);
+			_disposed = true;
```

Comment:
Why setting `_disposed` two times?
I think it would be enough to do it right after the `if (_disposed)` check.